### PR TITLE
[Docs] Random Fixes

### PR DESCRIPTION
**Pull request by noroadsleft**
## Description

Cleaned up some issues I found.

## Types of Changes

- [x] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/internals_gpio_control.md [hunk 1]
```diff
@@ -1,22 +1,86 @@
...
+
+
+### `readPin(pin)` :id=readpin
+
+Returns the level of the pin.
+
+| Architecture | Old Function                          |
+|--------------|---------------------------------------|
+| AVR          | `_SFR_IO8(pin >> 4) & _BV(pin & 0xF)` |
+| ChibiOS/ARM  | `palReadLine(pin)`                    |
```
Comment (fauxpark) [inline docs/internals_gpio_control.md]:
```suggestion
| ChibiOS/Arm  | `palReadLine(pin)`                    |
```

File: docs/internals_gpio_control.md [hunk 2]
```diff
@@ -1,22 +1,86 @@
...
+
+
+### `writePin(pin, level)` :id=writepin
+
+Set pin level, assuming it is an output.
+
+| Architecture | Old Function                                    |
+|--------------|-------------------------------------------------|
+| AVR          | `(level) ? PORTB \|= (1<<2) : PORTB &= ~(1<<2)` |
+| ChibiOS/ARM  | `(level) ? palSetLine(pin) : palClearLine(pin)` |
```
Comment (fauxpark) [inline docs/internals_gpio_control.md]:
```suggestion
| ChibiOS/Arm  | `(level) ? palSetLine(pin) : palClearLine(pin)` |
```

File: docs/internals_gpio_control.md [hunk 3]
```diff
@@ -1,22 +1,86 @@
...
+
+
+### `writePinLow(pin)` :id=writepinlow
+
+Set pin level as low, assuming it is an output.
+
+| Architecture | Old Function                                 |
+|--------------|----------------------------------------------|
+| AVR          | `PORTB &= ~(1<<2)`                           |
+| ChibiOS/ARM  | `palClearLine(pin)`                          |
```
Comment (fauxpark) [inline docs/internals_gpio_control.md]:
```suggestion
| ChibiOS/Arm  | `palClearLine(pin)`                          |
```

File: docs/internals_gpio_control.md [hunk 4]
```diff
@@ -1,22 +1,86 @@
...
+| ChibiOS/ARM  | `palSetLineMode(pin, PAL_MODE_OUTPUT_PUSHPULL)` |
+
+### `writePinHigh(pin)` :id=writepinhigh
+
+Set pin level as high, assuming it is an output.
+
+| Architecture | Old Function       |
+|--------------|--------------------|
+| AVR          | `PORTB \|= (1<<2)` |
+| ChibiOS/ARM  | `palSetLine(pin)`  |
```
Comment (fauxpark) [inline docs/internals_gpio_control.md]:
```suggestion
| ChibiOS/Arm  | `palSetLine(pin)`  |
```

File: docs/internals_gpio_control.md [hunk 5]
```diff
@@ -1,22 +1,86 @@
...
+| ChibiOS/ARM  | `palSetLineMode(pin, PAL_MODE_INPUT_PULLDOWN)` |
+
+### `setPinOutput(pin)` :id=setpinoutput
+
+Set pin as output.
+
+| Architecture | Old Function                                    |
+|--------------|-------------------------------------------------|
+| AVR          | `DDRB \|= (1<<2)`                               |
+| ChibiOS/ARM  | `palSetLineMode(pin, PAL_MODE_OUTPUT_PUSHPULL)` |
```
Comment (fauxpark) [inline docs/internals_gpio_control.md]:
```suggestion
| ChibiOS/Arm  | `palSetLineMode(pin, PAL_MODE_OUTPUT_PUSHPULL)` |
```

File: docs/internals_gpio_control.md [hunk 6]
```diff
@@ -1,22 +1,86 @@
...
+| ChibiOS/ARM  | `palSetLineMode(pin, PAL_MODE_INPUT_PULLUP)` |
+
+### `setPinInputLow(pin)` :id=setpininputlow
+
+Set pin as input with builtin pull-down resistor.
+
+| Architecture | Old Function                                   |
+|--------------|------------------------------------------------|
+| AVR          | N/A (Not supported on AVR)                     |
+| ChibiOS/ARM  | `palSetLineMode(pin, PAL_MODE_INPUT_PULLDOWN)` |
```
Comment (fauxpark) [inline docs/internals_gpio_control.md]:
```suggestion
| ChibiOS/Arm  | `palSetLineMode(pin, PAL_MODE_INPUT_PULLDOWN)` |
```

File: docs/internals_gpio_control.md [hunk 7]
```diff
@@ -1,22 +1,86 @@
...
+| ChibiOS/ARM  | `palSetLineMode(pin, PAL_MODE_INPUT)` |
+
+### `setPinInputHigh(pin)` :id=setpininputhigh
+
+Set pin as input with builtin pull-up resistor.
+
+| Architecture | Old Function                                 |
+|--------------|----------------------------------------------|
+| AVR          | `DDRB &= ~(1<<2); PORTB \|= (1<<2)`          |
+| ChibiOS/ARM  | `palSetLineMode(pin, PAL_MODE_INPUT_PULLUP)` |
```
Comment (fauxpark) [inline docs/internals_gpio_control.md]:
```suggestion
| ChibiOS/Arm  | `palSetLineMode(pin, PAL_MODE_INPUT_PULLUP)` |
```

File: docs/internals_gpio_control.md [hunk 8]
```diff
@@ -1,22 +1,86 @@
...
-
-## Advanced Settings
+### `setPinInput(pin)` :id=setpininput
+
+Set pin as input with high impedance (High-Z).
+
+| Architecture | Old Function                          |
+|--------------|---------------------------------------|
+| AVR          | `DDRB &= ~(1<<2)`                     |
+| ChibiOS/ARM  | `palSetLineMode(pin, PAL_MODE_INPUT)` |
```
Comment (fauxpark) [inline docs/internals_gpio_control.md]:
```suggestion
| ChibiOS/Arm  | `palSetLineMode(pin, PAL_MODE_INPUT)` |
```

File: docs/i2c_driver.md [hunk 9]
```diff
@@ -37,12 +42,12 @@ The following defines can be used to configure the I2C master driver.
 
 AVRs usually have set GPIO which turn into I2C pins, therefore no further configuration is required.
 
-## ARM
+## ARM :id=arm
```
Comment (fauxpark) [inline docs/i2c_driver.md]:
```suggestion
## Arm :id=arm
```

File: docs/i2c_driver.md [hunk 10]
```diff
@@ -37,12 +42,12 @@ The following defines can be used to configure the I2C master driver.
...
 
 For ARM the Chibios I2C HAL driver is under the hood.
```
Comment (fauxpark) [inline docs/i2c_driver.md]:
```suggestion
For Arm the Chibios I2C HAL driver is under the hood.
```

File: docs/i2c_driver.md [hunk 11]
```diff
@@ -37,12 +42,12 @@ The following defines can be used to configure the I2C master driver.
...
 This section assumes an STM32 MCU.
 
-### Configuration
+### Configuration :id=arm-configuration
 
 The configuration for ARM MCUs can be quite complex as often there are multiple I2C drivers which can be assigned to a variety of ports.
```
Comment (fauxpark) [inline docs/i2c_driver.md]:
```suggestion
The configuration for Arm MCUs can be quite complex as often there are multiple I2C drivers which can be assigned to a variety of ports.
```

**Comment by noroadsleft** [comment]
Personally I think the `Arm` rebranding is stupid and I don't like it. Maybe in a separate PR.

**Comment by noroadsleft** [comment]
@kymok @mtei @shelaf Naturally, I would discover an issue with the English I2C Master Driver doc right after you all worked on translating it to Japanese.

It's mostly formatting changes; the table at the top doesn't fit dimensionally with the new docs theme. I pulled the edits I made to the Japanese doc out because of the internal tracking that your contingent uses – the English doc's commit hash won't be known until this is merged, so there's no way to include it in the file.

**Comment by shelaf** [comment]
All right. We will track this PR.

**Comment by noroadsleft** [comment]
@shelaf https://github.com/qmk/qmk_firmware/blob/5784ddc752ff0c9f5f53b34f24b4de9d569ad1d8/docs/ja/i2c_driver.md was as far as I got before realizing it was potentially counter-productive to update it here. Feel free to use it as a baseline or copy the file outright; whatever works for you all.

**Comment by shelaf** [comment]
Thanks!

**Comment by mtei** [comment]
Commit hash cannot be controlled by the operator. But TAG can be specified arbitrarily.
Establishing a convention for TAG indicating the revision of the docs folder may help translators. For example Something like "docs/2020-03-08".

**Comment by noroadsleft** [comment]
> Establishing a convention for TAG indicating the revision of the docs folder may help translators. For example something like "docs/2020-03-08".

@mtei I like that idea quite a lot.

File: docs/_summary.md [hunk 12]
```diff
@@ -26,8 +26,8 @@
 
 * CLI
     * [Overview](cli.md)
-    * [Configuration](cli_configuration.md)
-    * [Commands](cli_commands.md)
+    * [Configuration](cli_dev_configuration.md)
+    * [Commands](cli.md#cli-commands)
```
Comment (mtei) [inline docs/_summary.md]: Adding `:id=` to the following lines in `docs/cli.md` will help translators:

```
# CLI Commands :id=cli-comands
```

Comment (noroadsleft) [reply docs/_summary.md]: Well spotted; will fix.

**Pull request merged**